### PR TITLE
Add useful options to prettier

### LINKED ISSUE #337: Feature suggestion: Automating wrapping markdown one-sentence-per-line
### Is Your Feature Request Related to a Problem? Please Describe

Add an automated way to wrap markdown files at 80 (@dstansby: maybe a bit more??) characters.

Suggested by @paddyroddy in https://github.com/UCL-ARC/python-tooling/pull/335#discussion_r1537991316.

### Describe the Solution You'd Like

An automated way to wrap markdown files in the pre-commit config.

### Describe Alternatives You've Considered

_No response_

### Additional Context

_No response_

### PR DESCRIPTION
**Pull request by paddyroddy**
Have added the flags and just ran over the repo - no manual changes

### CODE REVIEW

File: .github/workflows/links.yml [hunk 1]
```diff
@@ -26,7 +26,9 @@ jobs:
           # the whole of the project slug README since the links are
           # necessarily not real links - i.e. demos and to be filled by
           # cookiecutter values.
-          args: "--verbose --exclude https://forum.image.sc/ --exclude docs/pages --exclude-path '{{cookiecutter.project_slug}}/README.md'  -- ."
+          args:
+            "--verbose --exclude https://forum.image.sc/ --exclude docs/pages
+            --exclude-path '{{cookiecutter.project_slug}}/README.md'  -- ."
```
Comment (samcunliffe) [inline .github/workflows/links.yml:31]: No problem, but I'm surprised that this works. 🤷

File: .github/workflows/pages.yml [hunk 2]
```diff
@@ -48,7 +48,9 @@ jobs:
       - name: Build with Jekyll
         working-directory: docs
         # Outputs to the './_site' directory by default
-        run: bundle exec jekyll build --baseurl "${{ steps.pages.outputs.base_path }}"
+        run:
+          bundle exec jekyll build --baseurl "${{ steps.pages.outputs.base_path
+          }}"
```
Comment (samcunliffe) [inline .github/workflows/pages.yml:53]: 😞

File: .pre-commit-config.yaml [hunk 3]
```diff
@@ -29,6 +29,9 @@ repos:
     rev: v3.1.0
     hooks:
       - id: prettier
+        args:
+          - --prose-wrap=always
+          - --quote-props=as-needed
```
Comment (samcunliffe) [inline .pre-commit-config.yaml:34]: Solves #337

**Review by samcunliffe — APPROVED** [review]
LGTM.

Not sure I like the wrapping of yaml, but it seems to be valid so.... sure?

File: .github/workflows/links.yml (hunk 1 above)
Comment (paddyroddy) [reply .github/workflows/links.yml:31]: Yeah, I thought the same, but verified with [jless](https://jless.io)
image

**Pull request merged**